neural_compressor/onnxrt/utils/onnx_model.py

```diff
@@ -1,1249 +1,1259 @@
-            os.remove(os.path.join(os.path.dirname(self._model_path), external_data_location))
-        self.load_model_initializer_by_tensor()
-        convert_model_to_external_data(self._model, location=external_data_location)
-        # TODO : if init is already saved, skip write it
-        write_external_data_tensors(self._model, filepath=os.path.dirname(self._model_path))
-
-    def merge_split_models(self, to_merge_model):
-        """Merge two split model into final model."""
-        to_merge_model.write_external_data_to_new_location()
-        self.add_nodes([node for node in to_merge_model.nodes()])
-        self.add_initializers([init for init in to_merge_model.initializer()])
-        self.update()
-
-        # add new output
-        for output in to_merge_model.graph().output:
-            if output.name not in self.output():
-                self._model.graph.output.append(output)
-
-        # remove unused output
-        remove_output = []
-        for output in self._model.graph.output:
-            if output.name in to_merge_model.input():
-                remove_output.append(output)
-        for output in remove_output:
-            self._model.graph.output.remove(output)
-
-        # add new input
-        for input in to_merge_model.graph().input:
-            if (
-                input.name not in self.input()
-                and input.name not in self.output()
-                and input.name not in self.output_name_to_node.keys()
-            ):
-                self._model.graph.input.append(input)
-
-    def re_org_output(self, origin_output):
-        """Re-org output of merged model for layer-wise quantization."""
-        outputs = {}
-        tmp_remove = []
-        for output in self._model.graph.output:
-            outputs[output.name] = output
-            tmp_remove.append(output)
-
-        for output in tmp_remove:
-            self._model.graph.output.remove(output)
-
-        for out_name in origin_output:
-            self._model.graph.output.append(outputs[out_name])
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class for ONNX model."""
+
+import os
+import sys
+from pathlib import Path
+
+import onnx
+
+from neural_compressor.common import Logger
+from neural_compressor.onnxrt.utils.utility import MAXIMUM_PROTOBUF, find_by_name
+
+logger = Logger().get_logger()
+
+
+class ONNXModel:
```

Comment:
we should not introduce a new model class if it's not the subclass of existing framework model class.